src/pybind/mgr/dashboard/frontend/src/app/ceph/cephfs/cephfs-directories/cephfs-directories.component.ts

```diff
@@ -546,22 +547,44 @@ export class CephfsDirectoriesComponent implements OnInit, OnChanges {
           type: 'text',
           name: 'name',
           value: `${moment().toISOString(true)}`,
-          required: true
+          required: true,
+          validators: [this.validateValue.bind(this)]
         }
       ],
       submitButtonText: $localize`Create Snapshot`,
       onSubmit: (values: CephfsSnapshot) => {
-        this.cephfsService.mkSnapshot(this.id, path, values.name).subscribe((name) => {
+        if (!this.alreadyExists) {
+          this.cephfsService.mkSnapshot(this.id, path, values.name).subscribe((name) => {
+            this.notificationService.show(
+              NotificationType.success,
+              $localize`Created snapshot '${name}' for '${path}'`
+            );
+            this.forceDirRefresh();
+          });
+        } else {
           this.notificationService.show(
-            NotificationType.success,
-            $localize`Created snapshot '${name}' for '${path}'`
+            NotificationType.error,
+            $localize`Snapshot name '${values.name}' is already in use. Please use another name.`
           );
-          this.forceDirRefresh();
-        });
+        }
       }
     });
   }
 
+  validateValue(control: AbstractControl) {
+    const snapName = control.value;
+    let snapNames: string[] = [];
+    this.selectedDir.snapshots.forEach(name => {
+      snapNames.push(name['name']);
+    });
+    if(snapNames.includes(snapName)) {
+      return this.alreadyExists = true;
+    }
+    else {
+      return this.alreadyExists = false;
+    }
```

Comment:
?
```suggestion
    return snapNames.includes(snapName);
```

Reply:
@aaSharma14 i edited this after realizing my own mistake xD

Reply:
Yeah, I updated this.